tests/kernel/wave/wave_e2e_test.py

```diff
@@ -234,4 +234,4 @@ def test(
         run_config=config,
     ):
         test(a, b, c)
-        assert_allclose(ref, c, atol=0.1)
+        assert_allclose(ref, c, atol=0.15)

```

Comment:
Why did the tolerance go up? This is doing a sum so it should have remained the same? Also can you add a max e2e test (for which I am guessing you should match exactly?) And not for this PR, but would be nice to compare against iree to see whether there are any differences (there should be none?)

Reply:
great Q, yeap, it is unrelated to this PR. But there was one instance the error gets up to 0.125, since test generate random values. it was for a really large number along the magnitude of 84.872 to 84.747. So I set 0.15 just to make sure it doesn't break ever. I can also set manual seed instead. :)

Reply:
yeah let's set a manual seed so the tests results are reproducible.

Reply:
SG! let me push this up in a separate PR for better visibilityy and cleaner git commit history  :)

Reply:
Sounds good. Have you validated that the max reduction works e2e?

Reply:
yeap! I have another repo where I collect and test TK-Kernels, LMK push it up here as well

Reply:
Added test, and this one doesn't even need the atol modified. 😁

Reply:
nice! I think you can probably set to exact equality and it should still work because there is no loss in precision during max.

Reply:
ah shoot, missed this comment, I'll put it up in another PR fixing the reduce_sum test with manual seed. :)